pipelines/batch/src/main/java/com/google/fhir/analytics/FetchSearchPageFn.java

```diff
@@ -312,4 +341,48 @@ protected void processBundle(Bundle bundle, @Nullable Set<String> resourceTypes)
       }
     }
   }
+
+  private void processGoldenMerging(Bundle bundle) {
+    log.debug("Processing golden resources for Bundle");
+
+    bundle
+        .getEntry()
+        .removeIf(
+            entry -> {
+              Resource resource = entry.getResource();
+              if (mdmResourceList.contains(resource.fhirType())) {
+                String goldenResourceId =
+                    getGoldenResourceReference(resource.getId()).getReference();
+                // TODO: ACS-7652 filter invalid Golden Resources
+                return !resource.getId().equals(goldenResourceId);
+              }
+
+              return false;
+            });
+
+    for (BundleEntryComponent entry : bundle.getEntry()) {
+      Resource resource = entry.getResource();
+
+      Optional<StringType> subject =
+          fhirPath.evaluateFirst(resource, "subject.reference", StringType.class);
+      if (subject.isPresent()) {
+        resource.setProperty("subject", getGoldenResourceReference(subject.get().getValue()));
+        continue;
```

Comment:
Since I merged to `master` this also has the entire HTTP method as well. I don't think it's worth keeping around since it's a lot worse than the JDBC mode now, and would take even more `$mdm-query-links` queries to get it up to par.